source/geometries/Cigar.cc

```diff
@@ -0,0 +1,512 @@
+    fiber_ = new GenericWLSFiber(fiber_type_, true, true, fiber_diameter_, cigar_length_ + 7 * cm, true, coated_, this_coating, this_fiber, true);
+
+    // WORLD /////////////////////////////////////////////////
+
+    G4Material* world_mat = nullptr;
+
+    if (gas_ == "Ar") {
+      world_mat = materials::GAr(pressure_);
+      world_mat->SetMaterialPropertiesTable(opticalprops::GAr(1. / (68 * eV)));
+
+    } else if (gas_ == "Xe") {
+      world_mat = materials::GXe(pressure_);
+      // world_mat->SetMaterialPropertiesTable(opticalprops::GXe(pressure_, 273.15, 250, (1.0E9)));
+      world_mat->SetMaterialPropertiesTable(opticalprops::GXe());
+    } else {
+      G4Exception("[Cigar]", "Construct()",
+            FatalException, "Invalid gas, must be Ar or Xe");
+    }
+
+    G4Box* world_solid_vol =
+     new G4Box("WORLD", world_xy_/2., world_xy_/2., world_z_/2.);
+
+    G4LogicalVolume* world_logic_vol =
+      new G4LogicalVolume(world_solid_vol, world_mat, "WORLD");
+    world_logic_vol->SetVisAttributes(G4VisAttributes::GetInvisible());
+    GeometryBase::SetLogicalVolume(world_logic_vol);
+
+    // TEFLON PANELS ////////////////////////////////////////////
+    G4double panel_width = 2.5 * mm;
+    G4double extra_width = (5) * mm;
+    // G4Box* teflon_panel_top =
+    //   new G4Box("TEFLON_PANEL_TOP", cigar_width_ / 2 + extra_width + panel_width, panel_width / 2, cigar_length_ / 2);
+    G4Box* teflon_panel_top =
+      new G4Box("TEFLON_PANEL_TOP", cigar_width_ / 2 + extra_width + panel_width, panel_width / 2, cigar_length_ / 2);
+    G4Box* teflon_panel_side =
+      new G4Box("TEFLON_PANEL_SIDE", cigar_width_ / 2 + extra_width, panel_width / 2, cigar_length_ / 2);
+    G4Box* teflon_panel_close =
+      new G4Box("TEFLON_PANEL_CLOSE", cigar_width_ / 2 + extra_width+panel_width, cigar_width_ / 2 + extra_width+panel_width, panel_width / 2);
+    G4Material* teflon = G4NistManager::Instance()->FindOrBuildMaterial("G4_TEFLON");
+    teflon->SetMaterialPropertiesTable(opticalprops::PTFE());
+    G4LogicalVolume* teflon_logic_top =
+      new G4LogicalVolume(teflon_panel_top, teflon, "TEFLON");
+    G4LogicalVolume* teflon_logic_side =
+      new G4LogicalVolume(teflon_panel_side, teflon, "TEFLON");
+    G4LogicalVolume* teflon_logic_close =
+      new G4LogicalVolume(teflon_panel_close, teflon, "TEFLON_CLOSE");
+    teflon_logic_top->SetVisAttributes(nexus::White());
+    teflon_logic_side->SetVisAttributes(nexus::White());
+    teflon_logic_close->SetVisAttributes(nexus::White());
+
+
+
+
+
+    G4OpticalSurface* opsur_teflon =
+      new G4OpticalSurface("TEFLON_OPSURF", unified, groundteflonair, dielectric_metal);
+    opsur_teflon->SetMaterialPropertiesTable(opticalprops::PTFE());
+
+    new G4LogicalSkinSurface("TEFLON_OPSURF", teflon_logic_top, opsur_teflon);
+
+    G4VPhysicalVolume *teflon_top = new G4PVPlacement(0, G4ThreeVector(0, cigar_width_/2+panel_width/2+fiber_diameter_+extra_width/2+panel_width/2+fiber_diameter_/4,0),
```

Comment:
You don't need to define a variable for the placement, if you're not going to use it later. You can just write:

`new G4PVPlacement(...)`